### PR TITLE
qa/tests: added quincy-x suite initial draft

### PR DESCRIPTION
**Pull request by yuriw**
Signed-off-by: Yuri Weinstein <yweinste@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by yuriw** [comment]
@adk3798 @neha-ojha pls suggest regarding use of `ceph-adm` in this suite

**Comment by yuriw** [comment]
`CEPH_BRANCH=master; MACHINE_NAME=smithi; teuthology-suite -v -c $CEPH_BRANCH  -n 100 -m $MACHINE_NAME -s upgrade/quincy-x/parallel -k distro -e $CEPH_QA_EMAIL --suite-branch wip-yuriw-quincy-x-master -p 70 --force-priority  -l 1`

http://pulpito.ceph.com/teuthology-2022-03-14_16:32:03-upgrade:quincy-x:parallel-master-distro-default-smithi/

File: qa/suites/upgrade/quincy-x/stress-split/1-start.yaml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    conf:
+      osd:
+        #set config option for which cls modules are allowed to be loaded / used
+        osd_class_load_list: "*"
+        osd_class_default_list: "*"
+
+- cephadm.shell:
+    mon.a:
+      - ceph fs volume create foo
+      - ceph config set mon mon_warn_on_insecure_global_id_reclaim false --force
```
Comment (jdurgin) [inline qa/suites/upgrade/quincy-x/stress-split/1-start.yaml:20]: can remove these mon_warn lines, shouldn't be needed in quincy

**Review by jdurgin — APPROVED** [review]
looks good, just a couple things we can drop with quincy

File: qa/suites/upgrade/quincy-x/stress-split/1-start.yaml (hunk 1 above)
Comment (yuriw) [reply qa/suites/upgrade/quincy-x/stress-split/1-start.yaml:20]: removed two `mon_warn lines`

File: qa/suites/upgrade/octopus-x/parallel/1-tasks.yaml [hunk 2]
```diff
@@ -16,7 +16,7 @@ tasks:
         osd_class_default_list: "*"
     # deploy additional mons the "old" (octopus) way
     add_mons_via_daemon_add: true
-    avoid_pacific_features: true
+    avoid_quincy_features: true
```
[the remark this answers is not in the document]
Comment (yuriw) [reply qa/suites/upgrade/octopus-x/parallel/1-tasks.yaml:19]: This actually was an unintentional change as 1octopus-x1 is not in the scope of this PR, removed

**Comment by yuriw** [comment]
http://pulpito.front.sepia.ceph.com/teuthology-2022-03-16_17:34:28-upgrade:quincy-x:parallel-master-distro-default-smithi

http://pulpito.front.sepia.ceph.com/teuthology-2022-03-16_17:35:05-upgrade:quincy-x:stress-split-master-distro-default-smithi

File: qa/suites/upgrade/octopus-x/parallel/1-tasks.yaml [hunk 3]
```diff
@@ -16,7 +16,6 @@ tasks:
         osd_class_default_list: "*"
     # deploy additional mons the "old" (octopus) way
     add_mons_via_daemon_add: true
-    avoid_pacific_features: true
```
Comment (neha-ojha) [inline qa/suites/upgrade/octopus-x/parallel/1-tasks.yaml:19]: We don't need to test octopus any longer, just pacific-x and quincy-x.

File: qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+tasks:
+- install:
+    branch: pacific
+    exclude_packages:
+      - ceph-volume
+- print: "**** done install task..."
+- print: "**** done start installing pacific cephadm ..."
+- cephadm:
+    image: docker.io/ceph/daemon-base:latest-pacific
+    cephadm_branch: pacific
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml:10]: This should all be quincy.

File: qa/suites/upgrade/quincy-x/stress-split/1-start.yaml [hunk 5]
```diff
@@ -0,0 +1,120 @@
+tasks:
+- install:
+    branch: pacific
+    exclude_packages:
+      - ceph-volume
+
+- cephadm:
+    image: docker.io/ceph/daemon-base:latest-pacific
+    cephadm_branch: pacific
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/1-start.yaml:9]: This should also be quincy.

File: qa/suites/upgrade/quincy-x/stress-split/2-first-half-tasks/rbd-cls.yaml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+meta:
+- desc: |
+   run basic cls tests for rbd
+first-half-tasks:
+- workunit:
+    branch: octopus
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/2-first-half-tasks/rbd-cls.yaml:6]: why?

File: qa/suites/upgrade/quincy-x/stress-split/2-first-half-tasks/rbd-import-export.yaml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+meta:
+- desc: |
+   run basic import/export cli tests for rbd
+first-half-tasks:
+- workunit:
+    branch: octopus
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/2-first-half-tasks/rbd-import-export.yaml:6]: why?

File: qa/suites/upgrade/quincy-x/stress-split/3-stress-tasks/rbd-cls.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+meta:
+- desc: |
+   run basic cls tests for rbd
+stress-tasks:
+- workunit:
+    branch: octopus
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/3-stress-tasks/rbd-cls.yaml:6]: why?

File: qa/suites/upgrade/quincy-x/stress-split/3-stress-tasks/rbd-import-export.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+meta:
+- desc: |
+   run basic import/export cli tests for rbd
+stress-tasks:
+- workunit:
+    branch: octopus
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/3-stress-tasks/rbd-import-export.yaml:6]: why?

File: qa/suites/upgrade/quincy-x/stress-split/3-stress-tasks/rbd_api.yaml [hunk 10]
```diff
@@ -0,0 +1,10 @@
+meta:
+- desc: |
+   librbd C and C++ api tests
+stress-tasks:
+- workunit:
+     branch: octopus
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/3-stress-tasks/rbd_api.yaml:6]: why?

File: qa/suites/upgrade/quincy-x/stress-split/4-second-half-tasks/rbd-import-export.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+meta:
+- desc: |
+   run basic import/export cli tests for rbd
+second-half-tasks:
+- workunit:
+    branch: octopus
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/stress-split/4-second-half-tasks/rbd-import-export.yaml:6]: why?

File: qa/suites/upgrade/octopus-x/parallel/1-tasks.yaml (hunk 3 above)
Comment (yuriw) [reply qa/suites/upgrade/octopus-x/parallel/1-tasks.yaml:19]: removed `octopus-x`

File: qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+tasks:
+- install:
+    branch: quincy
+    exclude_packages:
+      - ceph-volume
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml:5]: don't we need this for quincy

File: qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+- print: "**** done install task..."
+- print: "**** done start installing quincy cephadm ..."
+- cephadm:
+    image: docker.io/ceph/daemon-base:latest-quincy
+    cephadm_branch: quincy
+    cephadm_git_url: https://github.com/ceph/ceph
```
Comment (neha-ojha) [inline qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml:11]: @adk3798 can you take a look at this?

**Review by neha-ojha — COMMENTED** [review]
The octopus-x tests included:

- stress-split-no-cephadm
- stress-split-erasure-code-no-cephadm
- rgw-multisite
-  parallel
- stress-split

The quincy-x tests will now have:

- parallel
- stress-split


There is a lot of extra test coverage (in terms of different kinds of configuration and thrashing) in octopus-x/stress-split-erasure-code-no-cephadm and stress-split-no-cephadm, which should be ported to the cephadm based upgrade tests. I noticed that the pacific-x upgrade tests are also missing those.

**Comment by yuriw** [comment]
https://pulpito.ceph.com/teuthology-2022-03-17_16:40:19-upgrade:quincy-x:parallel-master-distro-default-smithi

https://pulpito.ceph.com/teuthology-2022-03-17_16:41:06-upgrade:quincy-x:stress-split-master-distro-default-smithi

**Comment by djgalloway** [comment]
This is wrong:
```
  tasks:
  - cephadm:
      cephadm_branch: quincy
      cephadm_git_url: https://github.com/ceph/ceph
      conf:
        osd:
          osd_class_default_list: '*'
          osd_class_load_list: '*'
      image: docker.io/ceph/daemon-base:latest-quincy
```

We haven't pushed dev container images there for months.

File: qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml (hunk 13 above)
Comment (adk3798) [reply qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml:11]: image should probably be `quay.io/ceph/daemon:latest-quincy-devel` instead

Comment (yuriw) [reply qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml]: testing on this fix:

https://pulpito.ceph.com/teuthology-2022-03-17_17:37:21-upgrade:quincy-x:parallel-master-distro-default-smithi

https://pulpito.ceph.com/teuthology-2022-03-17_17:38:06-upgrade:quincy-x:stress-split-master-distro-default-smithi

Comment (yuriw) [reply qa/suites/upgrade/quincy-x/parallel/1-tasks.yaml]: http://pulpito.front.sepia.ceph.com/teuthology-2022-03-17_19:56:51-upgrade:quincy-x:parallel-master-distro-default-smithi
http://pulpito.front.sepia.ceph.com/teuthology-2022-03-17_18:23:54-upgrade:quincy-x:stress-split-master-distro-default-smithi

per @neha-ojha these tests show:

"http://pulpito.front.sepia.ceph.com/teuthology-2022-03-17_18:23:54-upgrade:quincy-x:stress-split-master-distro-default-smithi"

I think this is resolved and we will wait for a little before merging it with hope we will have the initial `reel` release additions

**Comment by yuriw** [comment]
jenkins retest this please

**Comment by ljflores** [comment]
Sorry, attempted a rebase and it went wrong.

**Comment by ljflores** [comment]
Alright, here is a successful rebase.

**Comment by ljflores** [comment]
jenkins test make check

**Comment by yuriw** [comment]
reef run:
`CEPH_QA_EMAIL="ceph-qa@ceph.io"; CEPH_BRANCH=wip-all-kickoff-r; MACHINE_NAME=smithi; teuthology-suite -v -c $CEPH_BRANCH -n 100 -m $MACHINE_NAME -s upgrade/quincy-x/parallel -k distro -e $CEPH_QA_EMAIL --suite-branch wip-yuriw-quincy-x-master -p 70 --force-priority -l 1 --suite-repo https://github.com/ceph/ceph.git`

http://pulpito.front.sepia.ceph.com/teuthology-2022-10-04_17:17:42-upgrade:quincy-x:parallel-wip-all-kickoff-r-distro-default-smithi

**Comment by ljflores** [comment]
@yuriw it's green!

**Comment by ljflores** [comment]
The branch needs to be rebased again though.

**Comment by ljflores** [comment]
@yuriw rebased it.

**Review by ljflores — APPROVED** [review]

**Comment by ljflores** [comment]
Oh, I see-- lots of conflicts after the Reef kickoff PR was merged. I'll need to check how much overlap there is. This PR can be closed if the Reef kickoff covers everything.

**Comment by yuriw** [comment]
run on main

http://pulpito.front.sepia.ceph.com:80/teuthology-2022-10-05_14:50:39-upgrade:quincy-x-main-distro-default-smithi/

I think we can close this PR

**Comment by ljflores** [comment]
All green from the Reef kickoff PR:
http://pulpito.front.sepia.ceph.com/lflores-2022-10-04_22:22:19-upgrade:quincy-x-wip-all-kickoff-r-distro-default-smithi/

**Pull request closed without merging**